plugins/modules/s3_object.py

```diff
@@ -1305,11 +1323,19 @@ def copy_object_to_bucket(module, s3, bucket, obj, encrypt, metadata, validate,
                 # with metadata that's provided in the request. The default value is 'COPY', therefore when user
                 # specifies a metadata we should set it to 'REPLACE'
                 params.update({"MetadataDirective": "REPLACE"})
-            s3.copy_object(aws_retry=True, **params)
+            response = s3.copy_object(aws_retry=True, **params)
+            changed |= response.get("VersionId") is not None or response.get("CopyObjectResult").get(
```

Comment:
If the destination bucket has versioning enabled, `VersionId` is given in the response, and hence the module flags a change, even though the new and old etags are the same.

Reply:
I think here we should report a change because idempotency is achieved when nothing has changed in the infrastructure, here a new version of the object has been created